girder/cli/serve.py

```diff
@@ -5,17 +5,23 @@
 
 from girder import _attachFileLogHandlers
 from girder.utility import server
+from girder.constants import PRODUCTION_MODE, DEVELOPMENT_MODE, TESTING_MODE
 
 
 @click.command(name='serve', short_help='Run the Girder server.', help='Run the Girder server.')
-@click.option('-t', '--testing', is_flag=True, help='Run in testing mode')
+@click.option('--dev', default=False, is_flag=True, help='Sets the server mode to development')
+@click.option('--mode', type=click.Choice([PRODUCTION_MODE, DEVELOPMENT_MODE, TESTING_MODE]),
+              default=PRODUCTION_MODE, show_default=True, help='Specify the server mode')
 @click.option('-d', '--database', default=cherrypy.config['database']['uri'],
               show_default=True, help='The database URI to connect to')
 @click.option('-H', '--host', default=cherrypy.config['server.socket_host'],
               show_default=True, help='The interface to bind to')
 @click.option('-p', '--port', type=int, default=cherrypy.config['server.socket_port'],
               show_default=True, help='The port to bind to')
-def main(testing, database, host, port):
+def main(dev, mode, database, host, port):
+    if (dev):
+        mode = DEVELOPMENT_MODE
```

Comment:
If both `--dev` and `--mode` are set, should we emit an error / warning?

Reply:
I think so. I'll add a click error to my next commit.